app/src/main/java/com/github/bgabriel998/softwaredevproject/general/SettingsMapActivity.java

```diff
@@ -0,0 +1,186 @@
+package com.github.bgabriel998.softwaredevproject.general;
+
+import android.annotation.SuppressLint;
+import android.app.Activity;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.os.Handler;
+import android.preference.PreferenceManager;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ProgressBar;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.util.Pair;
+
+import com.github.bgabriel998.softwaredevproject.R;
+import com.github.bgabriel998.softwaredevproject.map.OSMMap;
+import com.github.bgabriel998.softwaredevproject.points.DownloadTopographyTask;
+import com.github.bgabriel998.softwaredevproject.points.GeonamesHandler;
+import com.github.bgabriel998.softwaredevproject.points.POIPoint;
+import com.github.bgabriel998.softwaredevproject.points.Point;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.osmdroid.bonuspack.location.POI;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Activity that allows the user to select a point around which compute and
+ * download POIPoints and an elevation map for offline usage.
+ */
+public class SettingsMapActivity extends AppCompatActivity {
+
+    private OSMMap osmMap = null;
+
+    Button backButton;
+    Button okButton;
+
+    Activity thisActivity;
+
+    private Point selectedPoint;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_settings_map);
+
+        this.backButton = findViewById(R.id.settingsMapBackButton);
+        this.backButton.setOnClickListener(v -> this.finish());
+
+        this.okButton = findViewById(R.id.settingsMapOkButton);
+        this.okButton.setOnClickListener(v -> {
+            saveToJson();
+        });
+
+        this.thisActivity = this;
```

Comment:
**This variable might not be usefull.** 
A simple call to `this.finish()` at line 158 should do the job. 
If not then `((Activity) this).finish()` should work

Reply:
The reason I did it it's cause `this` is called inside a definition of a class. Are you sure `this` refers to the `SettingsMapActivity` in that case?